tests/test_visitors/test_ast/test_annotations/test_argument_annotations.py

```diff
@@ -68,6 +75,33 @@ def function(empty_arg: Literal[None]): ...
 def function(empty_arg: Union[Literal[None], Optional[int]]): ...
 """
 
+wrong_nested_literal_annotation = """
+def function(arg: Literal[Literal[1, 2], 3]): ...
+"""
+wrong_nested_union_annotation = """
+def function(arg: Union[Union[int, str], float]): ...
```

Comment:
Let's also try `Literal[Literal[Literal[1]]]` and the same for `Union`.

Reply:
Should `Literal[Literal[Literal[1]]]` raise 2 violations or 1? I feel like it should be treated as multiple violations. It would certainly make sense for cases like `Union[Union[Literal[Literal[1]], int]]` to show 2 violations. But it also seems a bit verbose for the first case to have 2 violations.  I'm going with multiple violations for now but would like to hear your opinion.

Reply:
I agree. 👍